src/cli/testRunner.ts

```diff
@@ -0,0 +1,187 @@
+    defaultConfig.projects = browserNames.map(browserName => {
+      return {
+        name: browserName,
+        use: { browserName },
+      };
+    });
+  }
+
+  const overrides = overridesFromOptions(opts);
+  if (opts.headed)
+    overrides.use = { headless: false };
+  const runner = new Runner(defaultConfig, overrides);
+
+  function loadConfig(configFile: string) {
+    if (fs.existsSync(configFile)) {
+      if (process.stdout.isTTY)
+        console.log(`Using config at ` + configFile);
+      const loadedConfig = runner.loadConfigFile(configFile);
+      if (('projects' in loadedConfig) && opts.browser)
+        throw new Error(`Cannot use --browser option when configuration file defines projects. Specify browserName in the projects instead.`);
+      return true;
+    }
+    return false;
+  }
+
+  if (opts.config) {
+    const configFile = path.resolve(process.cwd(), opts.config);
+    if (!fs.existsSync(configFile))
+      throw new Error(`${opts.config} does not exist`);
+    if (fs.statSync(configFile).isDirectory()) {
+      // When passed a directory, look for a config file inside.
+      if (!loadConfig(path.join(configFile, tsConfig)) && !loadConfig(path.join(configFile, jsConfig))) {
+        // If there is no config, assume this as a root testing directory.
+        runner.loadEmptyConfig(configFile);
+      }
+    } else {
+      // When passed a file, it must be a config file.
+      loadConfig(configFile);
+    }
+  } else if (!loadConfig(path.resolve(process.cwd(), tsConfig)) && !loadConfig(path.resolve(process.cwd(), jsConfig))) {
+    // No --config option, let's look for the config file in the current directory.
+    // If not, do not assume that current directory is a root testing directory, to avoid scanning the world.
+    throw new Error(`Configuration file not found. Run "npx playwright test --help" for more information.`);
+  }
+
+  process.env.FOLIO_JUNIT_OUTPUT_NAME = process.env.PLAYWRIGHT_JUNIT_OUTPUT_NAME;
+  process.env.FOLIO_JUNIT_SUITE_ID = process.env.PLAYWRIGHT_JUNIT_SUITE_ID;
+  process.env.FOLIO_JUNIT_SUITE_NAME = process.env.PLAYWRIGHT_JUNIT_SUITE_NAME;
+  process.env.FOLIO_JSON_OUTPUT_NAME = process.env.PLAYWRIGHT_JSON_OUTPUT_NAME;
+
+  const result = await runner.run(!!opts.list, args.map(forceRegExp), opts.project || undefined);
+  if (result === 'sigint')
+    process.exit(130);
+  process.exit(result === 'passed' ? 0 : 1);
+}
+
+function forceRegExp(pattern: string): RegExp {
+  const match = pattern.match(/^\/(.*)\/([gi]*)$/);
+  if (match)
+    return new RegExp(match[1], match[2]);
+  return new RegExp(pattern, 'g');
+}
+
+function overridesFromOptions(options: { [key: string]: any }): Config {
+  const shardPair = options.shard ? options.shard.split('/').map((t: string) => parseInt(t, 10)) : undefined;
+  return {
+    forbidOnly: options.forbidOnly ? true : undefined,
+    globalTimeout: options.globalTimeout ? parseInt(options.globalTimeout, 10) : undefined,
```

Comment:
Seems like it was not latest folio ToT since https://github.com/microsoft/folio/commit/4f81d520422bcb97968c6a30c14259ba0c134458 is missing.